turbo/stages/headerdownload/header_algos.go

```diff
@@ -571,9 +572,44 @@ func (hd *HeaderDownload) RequestMoreHeaders(currentTime uint64) (*HeaderRequest
 	return nil, penalties
 }
 
+func (hd *HeaderDownload) RequestMoreHeadersForPOS(currentTime uint64) *HeaderRequest {
+	hd.lock.Lock()
```

Comment:
If this method does not change anything inside `hd`, you should do `hd.lock.RLock()` and `hd.lock.RUnlock` instead